src/utils/errors.ts

```diff
@@ -50,24 +50,27 @@ export function getErrorMessage(
   unsupportedPerSource: Map<string, [string, ASTNode][]>,
   initialMessage: string,
 ): string {
-  let errorNum = 0;
-  const errorMsg = [...unsupportedPerSource.entries()].reduce(
-    (fullMsg, [filePath, unsopported]) => {
-      const content = fs.readFileSync(filePath, { encoding: 'utf8' });
-      const newMessage = unsopported.reduce((newMessage, [errorMsg, node]) => {
-        const errorCode = getSourceFromLocations(
-          content,
-          [parseSourceLocation(node.src)],
-          error,
-          4,
-        );
-        errorNum += 1;
-        return newMessage + `\n${error(`${errorNum}. ` + errorMsg)}:\n\n${errorCode}\n`;
-      }, `\nFile ${filePath}:\n`);
+  const errorData = Array.from(unsupportedPerSource.entries());
 
-      return fullMsg + newMessage;
-    },
-    error(initialMessage + '\n'),
-  );
-  return errorMsg;
+  const sourceErrors = errorData.map(([filePath, errors]) => {
+    const content = readFileSync(filePath, { encoding: 'utf8' });
+
+    return {
+      path: filePath,
+      errors: errors.map(([message, node]) => ({
+        message,
+        code: getSourceFromLocations(content, [parseSourceLocation(node.src)], error, 4),
+      })),
+    };
+  });
+
+  let errorId = 0;
+
+  return [
+    error(initialMessage),
+    sourceErrors.flatMap((source) => [
+      `File: ${source.path}:`,
+      ...source.errors.map((err) => `${error(`${++errorId}. ${err.message}`)}:\n\n${err.code}`),
```

Comment:
Actually, not sure something similar can be avoided in this case since we're doing a flat map including all errors. The index we'll be getting in `flatMap` is the index from the source array, and each `source` can contain multiple errors.

Reply:
Ohh, sorry I didn't see line 73 arrow function